common/motion_utils/src/marker/marker_helper.cpp

```diff
@@ -135,39 +135,27 @@ visualization_msgs::msg::MarkerArray createDeletedDeadLineVirtualWallMarker(
 visualization_msgs::msg::MarkerArray VirtualWallMarkerCreator::handleVirtualWallMarker(
   const std::vector<Pose> & poses, const std::string & module_name, const rclcpp::Time & now,
   int32_t id, create_wall_function function_create_wall_marker,
-  delete_wall_function function_delete_wall_marker,
-  std::vector<geometry_msgs::msg::Pose> & previous_virtual_wall_poses,
+  delete_wall_function function_delete_wall_marker, size_t & previous_virtual_walls_nb,
   const double longitudinal_offset)
 {
-  size_t id_to_create = id;
-  size_t id_to_delete = id;
   visualization_msgs::msg::MarkerArray wall_marker;
 
+  const auto max_id = id + static_cast<int32_t>(previous_virtual_walls_nb);
+
   if (poses.size() == 0) {
```

Comment:
poses.empty() is better?

Reply:
sorry 
when pose is empty, return should not be written. previuslly published pose should be deleted.
so this `if` part is not necessary